style_guide.md

```diff
@@ -0,0 +1,270 @@
+* Examples: `MyCoin`, `Position`, `PositionXY`.
+
+#### Event Names
+* Events must be named using the CapWords style. 
+* Event names will typically be singular.
+* Examples: `Deposit`, `Transfer`, `Approval`.
+
+#### Function Names
+* Functions other than constructors must use mixedCase. 
+  * `getBalance`, `transfer`, `verifyOwner`. 
+* Private or internal functions should use _underscoreMixedCase. 
+  * `_calculateBalance` , `_doTransfer`.
+
+#### Local and State Variable Names
+- Use mixedCase
+  * `totalSupply`, `balancesOf`, `creatorAddress`.
+- Variable names that refer to contracts are in the form: 
+  * `daiToken`, `yvDaiToken`, `sharesToken`.
+- Token amounts are to be referred as the contract variable, without `Token`, plus a word denoting the action undetaken with the token. In/Out will be the preferred words to denote tokens being transferred into/out of the contract. 
+  * `daiIn`, `daiOut`, `sharesMinted`, `yvDaiReceived`.
+
+#### Constant Names
+* Constants must be named with all capital letters with underscores separating words. 
+* Examples: `MAX_BLOCKS`, `CALLBACK_SUCCESS`, `CONTRACT_VERSION`.
+
+#### Mapping Names
+* Mappings should be mixed case with names that are plural. 
+* Examples: `deposits`, `balances`, `registeredAddresses`.
+
+#### Modifier Names
+* Use mixedCase. 
+* Examples: `onlyOwner`, `onlyAfter`, `onlyDuringThePreSale`.
+
+#### Enums
+* Enums, in the style of simple type declarations, must be named using the CapWords style. 
+* Examples: `TokenGroup`, `Frame`, `HashStyle`.
+
+
+## Code Comments
+
+#### NatSpec
+* Add NatSpec in a sensible fashion for contracts, functions and state variables[^2].
+
+#### In-line commenting
+* Every line of assembly should be clearly documented, for example, unchecked.
+* Useful for clarifying anything that is surprising, complicated, or risky.
+> Note: if you need to add a comment to explain what is that you are doing, take a moment to consider if you should do it differently.
+
+*** 
+<br>
+
+</br>
+
+# Other Points
+
+### uint over uint256
+* While both are equivalent, uint256 should always be used over uint.
+
+
+###  TransferHelper.sol
+* TransferHelper library(from yield-utils-v2) and its safe transfer methods should be used when ERC20 is involved, over the native methods.[^3]
+
+
+### Use of underscore
+
+In general, avoid the use of `_` in variable or fn names. There are only two exceptions:
+
+1. If the var or fn name shadows an existing name, then use a trailing underscore _
+Example: 
+```solidity
+uint256 public someNumber;
+
+function constructor(uint256 someNumber_) {
+   someNumber = someNumber_;
+}
+```
+
+2. If the var or fn is of visibility private or internal, use a preceding _
+
+```solidity
+uint256 internal _internalVar;
+
+function _internalCheck() internal {}
+}
+```
+
+
+
+### Compact Strings
+
+Ensure that your strings do not exceed **32 characters**. For example, in `require` statements. 
+
+**Explanation:**
```

Comment:
really great notes...but not sure if they belong here

Reply:
yeah, i'm still iffy about it myself. perhaps thrown in together with the other gas opt stuff as a separate doc. 

I think first we ought to come to a strict definition/scope for each doc (style guide, yieldcurity) - from there it would be easier to decide organization and if we want to spin-off another doc for gas opt.